src/app/clusters/level-control/level-control.cpp

```diff
@@ -950,14 +950,21 @@ static void moveHandler(app::CommandHandler * commandObj, const app::ConcreteCom
                         app::DataModel::Nullable<uint8_t> rate, chip::Optional<BitMask<OptionsBitmap>> optionsMask,
                         chip::Optional<BitMask<OptionsBitmap>> optionsOverride)
 {
-    EndpointId endpoint = commandPath.mEndpointId;
-    CommandId commandId = commandPath.mCommandId;
-
-    EmberAfLevelControlState * state = getState(endpoint);
     Status status;
-    app::DataModel::Nullable<uint8_t> currentLevel;
     uint8_t difference;
+    EmberAfLevelControlState * state;
+    app::DataModel::Nullable<uint8_t> currentLevel;
 
+    EndpointId endpoint = commandPath.mEndpointId;
+    CommandId commandId = commandPath.mCommandId;
+    // Validate the received rate and moveMode first.
+    if ((!rate.IsNull() && (rate.Value() == 0)) || moveMode == MoveModeEnum::kUnknownEnumValue)
```

Comment:
```suggestion
    if (rate.ValueOr(1) == 0) || moveMode == MoveModeEnum::kUnknownEnumValue)
```

Although I believe even `rate == 0` may actually do the right thing.

Reply:
Yes, `rate == 0` should do the right thing.

Reply:
Is it new that nullable variables have a direct the == operator to the value? Anyway I'll test/change that then

Reply:
So rate==0 errors out with
`error: ambiguous overload for 'operator==' (operand types are 'chip::app::DataModel::Nullable<unsigned char>' and 'int')`

casting the 0 as uin8_t here works. I don't know if there is a cleaner way to ensure the == operator is with the proper type.

Reply:
> Is it new that nullable variables have a direct the == operator to the value?

Yes.  That said, the fact that for integers it's got those overload issues is annoying.  :(